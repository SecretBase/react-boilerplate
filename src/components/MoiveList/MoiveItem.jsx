// @flow
import React, {PureComponent} from 'react'
import styled from 'styled-components'
import {lighten} from 'polished'

export type Moive = {
  name: string,
  emoji: string,
  className?: string
}

type Props = Moive

const PaleText = styled.span`
  font-size: 16px;
  color: ${lighten(0.1, 'pink')};
  &::selection {
    background-color: #000;
  }
`

class MoiveItem extends PureComponent<Props> {
  render () {
    return (
      <li className={this.props.className}>{this.props.emoji} <PaleText>{this.props.name}</PaleText></li>
    )
  }
}

export default styled(MoiveItem)`
  list-style-type: none;
  margin-bottom: 16px;
  font-size: 24px;
  color: #414042;
`
