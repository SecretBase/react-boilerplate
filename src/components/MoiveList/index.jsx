// @flow
import React, {PureComponent} from 'react'
import styled from 'styled-components'
import MoiveItem from './MoiveItem'
import type {Moive} from './MoiveItem'

export type Moives = Array<Moive>

type Props = {
  moives: Moives,
  className: string
}

class MoiveList extends PureComponent<Props> {
  render () {
    return (
      <ul className={this.props.className}>
        {this.props.moives.map(moive => <MoiveItem key={moive.name} {...moive} />)}
      </ul>
    )
  }
}

export default styled(MoiveList)`
  width: 980px;
  padding: 0;
  margin: auto;
`
