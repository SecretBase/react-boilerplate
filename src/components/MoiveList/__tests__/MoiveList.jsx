// @flow
import React from 'react'
import {shallow} from 'enzyme'

import MoiveList from '../index'

describe('MoiveList component', () => {
  it('should render correctly', () => {
    const props = {
      moives: [
        {
          name: 'hello',
          emoji: 'world'
        }
      ]
    }
    const wrapper = shallow(
      <MoiveList {...props} />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
