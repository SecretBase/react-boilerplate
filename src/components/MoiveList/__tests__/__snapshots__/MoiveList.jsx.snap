// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoiveList component should render correctly 1`] = `
<MoiveList
  className="sc-htpNat flqtFs"
  moives={
    Array [
      Object {
        "emoji": "world",
        "name": "hello",
      },
    ]
  }
/>
`;
